Amplify/Categories/DataStore/Query/QueryPaginationInput.swift

```diff
@@ -0,0 +1,38 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// A simple struct that holds pagination information that can be applied queries.
+public struct QueryPaginationInput {
+
+    /// The default page size
+    public static var defaultLimit: UInt = 100
+
+    public let page: UInt
+    public let limit: UInt
+
+}
+
+extension QueryPaginationInput {
+
+    public static func page(_ page: UInt,
+                            limit: UInt = QueryPaginationInput.defaultLimit) -> QueryPaginationInput {
+        return QueryPaginationInput(page: page, limit: limit)
```

Comment:
Hmm.. in the case where we have `page(0, limit: 0)` -- we could throw an exception.. or log some sort of error.. or.. just do nothing.
Seems like nothing could be okay -- what do you think?

Also, do we care to support an 'unlimited' parameter? (e.g. they want to specify a page, but specify an unlimited parameter)

Reply:
> Hmm.. in the case where we have page(0, limit: 0) -- we could throw an exception.. or log some sort of error.. or.. just do nothing.
Seems like nothing could be okay -- what do you think?

I'm willing to allow customers to do that if they want, like databases usually allow.

> Also, do we care to support an 'unlimited' parameter? (e.g. they want to specify a page, but specify an unlimited parameter)

How does a page is defined without knowing its size? That could work in case of an `offset` is used instead of a `page`, but since we  decided to calculate the offset based on the `page` and `limit`, both are needed (i.e. what range page 2 belongs to if we don't know the page size?).

Reply:
regarding page(0, limit:0) -- i was worried about unintentionally sending those values in.. but.. probably something we don't need to spend time debating.. i think it's a case where customers just need to be careful about what they're doing. :).

Regarding .unlimited:
Oops brain fart: I was thinking they might call page(0, limit:100), and then decide that based on those first 100 records, to then call page(1, limit: unlimited) -- but that doesn't make sense :)